Remove explicit port binding in Azurite container

ChlorUpload opened the pull request:
Fixes #875.

Switched from explicit port binding to automatic assignment to avoid port conflicts.

ChlorUpload commented on packages/modules/azurite/src/azurite-container.ts:
```diff
@@ -119,11 +89,7 @@ export class AzuriteContainer extends GenericContainer {
-      { container: QUEUE_PORT, host: this.queuePort },
-      { container: TABLE_PORT, host: this.tablePort }
-    );
+    this.withCommand(command).withExposedPorts(BLOB_PORT, QUEUE_PORT, TABLE_PORT);
```
If developers are allowed to use withExposedPort(), it becomes difficult to create the necessary endpoints as the port order for blob, queue, and table can be set arbitrarily. 

Appreciating the existing implementation, the withBlobPort, withQueuePort, and withTablePort functions were modified to accept PortWithOptionalBinding. By using hasHostBinding, it distinguishes whether the port is custom or automatically mapped. Could you review this again? 🙏

cristianrgreco commented:
One test failing:

```
FAIL packages/modules/azurite/src/azurite-container.test.ts (38.5 s)
  Azurite
    ✓ should upload and download blob with default credentials (10494 ms)
    ✓ should add to queue with default credentials (1322 ms)
    ✓ should add to table with default credentials (1306 ms)
    ✓ should be able to specify accountName and accountKey (1356 ms)
    ✕ should be able to specify custom ports (944 ms)
    ✕ should be able to use in-memory persistence (18332 ms)

  ● Azurite › should be able to specify custom ports

    expect(received).toBe(expected) // Object.is equality

    Expected: 13000
    Received: 32783

      125 |       .start();
      126 |
    > 127 |     expect(container.getBlobPort()).toBe(blobPort);
          |                                     ^
      128 |     expect(container.getQueuePort()).toBe(queuePort);
      129 |     expect(container.getTablePort()).toBe(tablePort);
      130 |

      at Object.<anonymous> (src/azurite-container.test.ts:127:37)

  ● Azurite › should be able to use in-memory persistence

    RestError: connect ECONNREFUSED ::1:32786

      at ClientRequest.<anonymous> (../../../node_modules/@azure/core-rest-pipeline/src/nodeHttpClient.ts:223:11)
```

ChlorUpload commented:
- Fixed test 'Azurite › should be able to specify custom ports'
- In the test 'Azurite › should be able to use in-memory persistence', `container.refresh()` now binds a new ports. So, I modified the test to create an azurite client from a new connection string including the new ports.
